syntax = "proto2";
package pidb;
option cc_generic_services = true;

message PiDBRequest{
	required string key = 1;
	optional string value = 2;
}

message PiDBResponse{
	required bool success = 1;
	optional string old_value = 2;
	optional string new_value = 3;
	optional string redirect = 4;
}

service PiDBService {
	rpc write(PiDBRequest) returns (PiDBResponse);
	rpc read(PiDBRequest) returns (PiDBResponse);
}
